src/backend/distributed/planner/distributed_planner.c

```diff
@@ -864,9 +894,17 @@ CreateDistributedPlan(uint64 planId, Query *originalQuery, Query *query, ParamLi
  * currently only implements some optimizations for intermediate result(s) pruning.
  */
 static void
-FinalizeDistributedPlan(DistributedPlan *plan, Query *originalQuery)
+FinalizeDistributedPlan(DistributedPlan *plan, Query *originalQuery,
+						bool fastPathRouterQuery)
```

Comment:
Could the fastPathRouterQuery by a property of the DistributedPlan to avoid cluttering the signatures?

Reply:
I think it makes sense, but keeping it as a part of `restrictionContext` still useful to pass the information during the planning.

Does that work if I keep it in both places? I think having it on the `distributedPlan` can help me in the next PRs (at least to have extra asserts etc)

Reply:
Yes, keeping in both places seems fine.